src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,221 @@
+      `Can't get the node url as the he node (${nodeId}) is undefined.`
+    );
+  }
+
+  let pid = agent.pid;
+  let hostname = agent.hostname;
+  return `//${hostname}:${pid.substring(pid.lastIndexOf(":") + 1)}`;
+}
+
+function getExecutorDirectory(agentId, frameworkId, taskId) {
+  var agentState = MesosStore.getState(agentId);
+
+  if (agentState == null) {
+    throw new Error(
+      `Unable get the executor directory as the agent (${agentId})
+      state is undefined.`
+    );
+  }
+
+  function matchFramework(framework) {
+    return frameworkId === framework.id;
+  }
+
+  let framework =
+    agentState.frameworks.find(matchFramework) ||
+    agentState.completed_frameworks.find(matchFramework);
+
+  if (!framework) {
+    throw new Error(
+      `Can't get the executor directory as the Framework (${frameworkId})
+       does not exist on the given agent (${agentId}).`
+    );
+  }
+
+  function matchExecutor(executor) {
+    return taskId === executor.id;
+  }
+
+  var executor =
+    framework.executors.find(matchExecutor) ||
+    framework.completed_executors.find(matchExecutor);
+
+  if (executor == null) {
+    throw new Error(
+      `Unable to get the executor directory as the task (${taskId})
+      does not exists.`
+    );
+  }
+
+  return executor.directory;
+}
+
+function resolveTaskFileRequests() {
+  var info = InfoStore.info;
+
+  // Get the marathon config, to retrieve the mesos leader (master)  url
+  if (!info.hasOwnProperty("marathon_config")) {
+    InfoActions.requestInfo();
+    return;
+  }
+
+  // Get master sate, We need it to parse the agent/node url
+  if (!MesosStore.getState(MASTER_ID)) {
+    MesosActions.requestState(MASTER_ID,
+      info.marathon_config.mesos_leader_ui_url.replace(/\/?$/, "/master"));
+    return;
+  }
+
+  // Check all requests, request needed data or remove them from the que
+  taskFileRequestQue.forEach((request, index) => {
+
+    var taskId = request.taskId;
+    var agentId = request.agentId;
+
+    // Get node/agent sate. This is needed to actually retrieve
+    // the executor directory.
+    if (!MesosStore.getState(agentId)) {
+      try {
+        MesosActions.requestState(agentId, getNodeUrl(agentId));
+      } catch (error) {
+        // Invalidate underlying data and start over
+        invalidateState(MASTER_ID);
+        resolveTaskFileRequests();
+      }
+      return;
+    }
+
+    // Request the files if not present
+    if (!MesosStore.getTaskFiles(taskId)) {
+      try {
+        MesosActions.requestFiles(taskId,
+          getNodeUrl(agentId),
+          getExecutorDirectory(agentId, info.frameworkId, taskId));
+      } catch (error) {
+        // Invalidate underlying data and start over
+        invalidateState(agentId);
+        resolveTaskFileRequests();
+      }
+      return;
+    }
+
+    // Everything is fine, we have the file, let's remove it from the que
+    taskFileRequestQue.splice(index, 1);
+  });
+}
+
+AppDispatcher.register(function (action) {
```

Comment:
In generally we try to emit all events from this place.

This means the private store functions are encapsulated, stands for themselves and don't emit directly from the store. They return values if necessary. So at the in the ```case``` you could emit. That holds the things much more at one place and is better readable.